src/utils/utils.ts

```diff
@@ -78,7 +78,8 @@ export function isPlainObject(value) {
     return proto === Object.prototype || proto === null
 }
 
-export function objectAssign(...objs: Object[]): Object
+export function objectAssign<T extends object>(target: { [key: string]: never }, clonedSource: T, ...sources: (Partial<T> & object)[]): T;
```

Comment:
`{ [key: string]: never }` why not just `{}`?

Reply:
Also `object` is `function` too

Reply:
A special treatment of an empty object.

`{}` in a signature would allow any object that implements `{}` (also known as any object).

`{[key: string]: never}` matches only `{}`

Reply:
Oh, nice!